provider-middleware/brightspace_data.go

```diff
@@ -49,15 +55,116 @@ type BrightspaceEnrollment struct {
+		imgBytes, err = getDefaultImg()
+	case http.StatusOK:
+		imgBytes, err = io.ReadAll(response.Body)
+	case http.StatusForbidden:
+		log.Warnf("status of forbidden returned, using default Brightspace image, url is %v", courseImageUrl)
+		imgBytes, err = getDefaultImg()
+	}
+	if err != nil {
+		imgPath = ""
+	} else {
+		imgPath, err = UploadBrightspaceImage(imgBytes, id)
+		if err != nil {
+			log.Errorf("error during upload of Brightspace image to UnlockEd, id used was: %v error is %v", id, err)
+			imgPath = ""
+		}
+	}
+	course := models.Course{
+		ProviderPlatformID:      srv.ProviderPlatformID,
+		ExternalID:              bsCourse.OrgUnitId,
+		Name:                    bsCourse.Name,
+		OutcomeTypes:            "completion",
+		ThumbnailURL:            imgPath,
+		Type:                    "fixed_enrollment",                             //open to discussion
+		Description:             "Brightspace Managed Course: " + bsCourse.Name, //WIP
+		TotalProgressMilestones: uint(0),                                        //WIP
+		ExternalURL:             srv.BaseURL,                                    //WIP
+	}
+	return &course
 }
 
-func (srv *BrightspaceService) GetPluginId(pluginName string) (string, error) {
+func UploadBrightspaceImage(imgBytes []byte, bsCourseId string) (string, error) {
+	filename := "image_brightspace" + "/" + bsCourseId + ".jpg"
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", filename)
+	if err != nil {
+		log.Errorf("error creating form file using file %v, error is %v", filename, err)
+		return "", err
+	}
+	if _, err = part.Write(imgBytes); err != nil {
+		log.Errorf("error writing bytes to mulitpart form, error is %v", err)
+		return "", err
+	}
+	err = writer.Close()
+	if err != nil {
+		log.Errorf("error closing file, error is %v", err)
+		return "", err
+	}
+	uploadEndpointUrl := os.Getenv("APP_URL") + "/upload"
+	request, err := http.NewRequest(http.MethodPost, uploadEndpointUrl, body)
+	if err != nil {
+		log.Errorf("error creating new POST request to url %v and error is: %v", uploadEndpointUrl, err)
+		return "", err
+	}
+	request.Header.Set("Content-Type", writer.FormDataContentType())
+	request.Header.Set("Content-Length", fmt.Sprintf("%d", len(body.Bytes())))
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		log.Errorf("error executing POST request to url %v and error is: %v", uploadEndpointUrl, err)
+		return "", err
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("server returned non-OK status: %s", response.Status)
+	}
+	urlRes := struct {
+		Data struct {
+			Url string `json:"url"`
+		}
+		Message string `json:"message"`
+	}{}
+	err = json.NewDecoder(response.Body).Decode(&urlRes)
+	if err != nil {
+		return "", err
+	}
+	return urlRes.Data.Url, nil
+}
+
+func getDefaultImg() ([]byte, error) {
```

Comment:
In the case that the download fails, can we have this just set the ThumbnailURL to `/brightspace.png` which we can keep in the /frontend/public directory, so we don't have to read and upload this default image each time?

Reply:
Yeah, much more efficient.  I'll make the change in the upcoming PR.